hasher-matcher-actioner/hmalib/lambdas/api/action_rules_api.py

```diff
@@ -0,0 +1,165 @@
+# Copyright (c) Facebook, Inc. and its affiliates. All Rights Reserved
+
+import bottle
+import typing as t
+import sys
+
+from botocore.exceptions import ClientError
+from bottle import response
+from dataclasses import dataclass, field
+from hmalib.common.logging import get_logger
+from .middleware import jsoninator, JSONifiable, DictParseable
+from hmalib.common.config import HMAConfig
+from hmalib.common import config as hmaconfig
+from hmalib.common.evaluator_models import ActionRule
+from hmalib.common.classification_models import Label, ActionLabel
+
+
+logger = get_logger(__name__)
+
+
+@dataclass
+class ActionRulesAPIRequest(DictParseable):
+    action_rule: ActionRule
+
+    @classmethod
+    def from_dict(cls, d: dict) -> "ActionRulesAPIRequest":
+        ar = ActionRule.from_aws(d["action_rule"])
+        logger.info("Deserialized ActionRule: %s", ar)
+        return cls(ar)
+
+
+@dataclass
+class ActionRulesAPIResponse(JSONifiable):
+    error_message: str
+    action_rules: t.List[ActionRule] = field(default_factory=list)
+
+    def to_json(self) -> t.Dict:
+        return {
+            "error_message": self.error_message,
+            "action_rules": [action_rule.to_aws() for action_rule in self.action_rules],
+        }
+
+
+def handle_unexpected_error(e: Exception):
+    logger.error("Unexpected error: %s", sys.exc_info()[0])
+    logger.exception(e)
+    response.status = 500
+
+
+def get_action_rules_api(hma_config_table: str) -> bottle.Bottle:
+    # The documentation below expects prefix to be '/action-rules/'
+    action_rules_api = bottle.Bottle()
+    HMAConfig.initialize(hma_config_table)
+
+    @action_rules_api.get("/", apply=[jsoninator])
+    def get_action_rules() -> ActionRulesAPIResponse:
+        """
+        Returns all action rules.
+        """
+        error_message = ""
+        action_rules = []
+
+        try:
+            action_rules = ActionRule.get_all()
+            logger.info("action_rules: %s", action_rules)
+        except Exception as e:
+            error_message = "Unexpected error."
+            handle_unexpected_error(e)
+
+        return ActionRulesAPIResponse(error_message, action_rules)
+
+    @action_rules_api.post("/", apply=[jsoninator(ActionRulesAPIRequest)])
+    def create_action_rule(
+        request: ActionRulesAPIRequest,
+    ) -> ActionRulesAPIResponse:
+        """
+        Creates an action rule.
+        """
+        logger.info("request: %s", request)
+        error_message = ""
+
+        try:
+            hmaconfig.create_config(request.action_rule)
+        except ClientError as e:
+            # TODO this test for "already exists" should be moved to a common place
```

Comment:
+1 just a helper with _if_exist in the name is something I added for privacy group 'configs'